lib/mayaUsd/ufe/UsdUIInfoHandler.cpp

```diff
@@ -106,14 +157,93 @@ std::string UsdUIInfoHandler::treeViewIcon(const Ufe::SceneItem::Ptr& item) cons
 		{"Volume",				"out_USD_Volume.png"}
 	};
 
+#if UFE_PREVIEW_VERSION_NUM >= 2024
+	Ufe::UIInfoHandler::Icon icon;		// Default is empty (no icon and no badge).
+#endif
+
 	const auto search = supportedTypes.find(item->nodeType());
 	if (search != supportedTypes.cend()) {
+#if UFE_PREVIEW_VERSION_NUM >= 2024
+		icon.baseIcon = search->second;
+#else
 		return search->second;
+#endif
 	}
 
+#if UFE_PREVIEW_VERSION_NUM >= 2024
+	// Check if we have any composition meta data - if yes we display a special badge.
+	UsdSceneItem::Ptr usdItem = std::dynamic_pointer_cast<UsdSceneItem>(item);
+	if (usdItem)
+	{
+		// Variants
+		if (!usdItem->prim().GetVariantSets().GetNames().empty())
+		{
+			icon.badgeIcon = "out_USD_CompArcBadgeV.png";
+			icon.pos = Ufe::UIInfoHandler::LowerRight;
+		}
+		else
+		{
+			// Composition related metadata.
+			static const std::vector<PXR_NS::TfToken> compKeys = {
+				PXR_NS::SdfFieldKeys->References, SdfFieldKeys->Payload, SdfFieldKeys->InheritPaths, SdfFieldKeys->Specializes};
+			for (const auto& k : compKeys)
+			{
+				if (usdItem->prim().HasMetadata(k))
+				{
+					icon.badgeIcon = "out_USD_CompArcBadge.png";
+					icon.pos = Ufe::UIInfoHandler::LowerRight;
+					break;
+				}
+			}
```

Comment:
Otherwise look and see if we have any of the meta data types: references, payload, inheritpaths, specializes. If yes, then add the composition arc badge icon.